### ISSUE TITLE
onnxruntime-gpu but GPU dependencies are not loaded

### ISSUE THREAD

**Issue by SuperSecureHuman** [huggingface/optimum#1402]
### System Info

```shell
Optimum 1.13.1
Python 3.9


pip list | grep onnx                                                                                                                                                                                                                                          
onnx                      1.14.1
onnxruntime-gpu           1.16.0
```


### Who can help?

@JingyaHuang @ech


### Reproduction (minimal, reproducible, runnable)

Trying to use ORT with CUDA raises error - 

`onnxruntime-gpu` is installed, but GPU dependencies are not loaded. It is likely there is a conflicting install between `onnxruntime` and `onnxruntime-gpu`. Please install only `onnxruntime-gpu` in order to use CUDAExecutionProvider."

While

![image](https://github.com/huggingface/optimum/assets/88489071/56d44139-a29f-4ba1-9467-ecf1e976f7ae)

Clearly, only ort-gpu is installed

Same issue can be reproduced in colab

`%pip install optimum[onnxruntime-gpu]`


`ORTModelForSeq2SeqLM.from_pretrained(.... providers=["CUDAExecutionProvider"], use_io_binding=True)`


### Expected behavior

No error, expecting to run ort pipeline on GPU.

### Workaround

Currently I am commenting out 

https://github.com/huggingface/optimum/blob/7fc27f6747e692fed45acc583e153d45541bc2e7/optimum/onnxruntime/utils.py#L214

Now the pipeline runs on GPU without any issues

**Comment by fxmarty** [comment]
Thank you @SuperSecureHuman, I can reproduce the issue. It appears `onnxruntime-gpu==1.16.0` does not fill `_ld_preload.py` like it used to. I will patch!

**Comment by SuperSecureHuman** [comment]
I am not very familiar with onnx internals.

But from my understanding, if something is there in the available providers, it should work right?

Why is the check needed?

**Comment by fxmarty** [comment]
The check is "needed" because ORT falls back to `CPUExecutionProvider` without any error in case the CUDA / TRT install is wrong (or in case `onnxruntime` was installed after `onnxruntime-gpu`), see: https://onnxruntime.ai/docs/execution-providers/#use-execution-providers. We had a few issues where users were confused because of this (e.g. ORT was using CPU EP although specifying `CUDAExecutionProvider`, or failing although `onnxruntime-gpu` was installed due to a double onnxruntime/onnxruntime-gpu install, etc.) so I implemented hard checks with meaningful errors.

For the record, here is how `onnxruntime/capi/_ld_preload.py` used to look like with `onnxruntime-gpu` (while empty for `onnxruntime`). It appears ORT 1.16.0 changed this.
![image](https://github.com/huggingface/optimum/assets/9808326/7b50c337-18f5-4df1-b36d-4d6d93061998)

Fixed in https://github.com/huggingface/optimum/pull/1403, will publish a patch today!
